references/kvm-admin-api/apiproxy/resources/jsc/JS-IAMPermission.js

```diff
@@ -0,0 +1,14 @@
+var reqMethod = context.getVariable("request.verb");
+var iamPermission = "";
+
+if(reqMethod === "POST")
+    iamPermission = "create";
+else if(reqMethod === "GET")
+    iamPermission = "list";
+else if(reqMethod === "DELETE")
+    iamPermission = "delete";
+else
+    iamPermission = "noop";
```

Comment:
note:`apigee.keyvaluemaps.noop` will always fail with:

```json
{
  "error": {
    "code": 400,
    "message": "Permission apigee.keyvaluemaps.noop is not valid for this resource.",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.RequestInfo",
        "requestId": "xxxx"
      }
    ]
  }
}
```

Reply:
Any unrecognized verb will trigger the 404 fallback in conditional flows. 
The noop is a little bit of a paranoid measure for future extensions. The block is unreachable for PUT requests.